echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/PipelineTriggerAction.groovy

```diff
@@ -27,21 +29,33 @@ import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
 class PipelineTriggerAction extends ActionSupport {
-    static final Logger LOGGER = LoggerFactory.getLogger(PipelineTriggerAction)
-
-    @Override
-    void execute(Context context, Execution execution) throws Exception {
-        try {
-            def pipelineInitiator = (PipelineInitiator) ActionDependencies.getBean(PipelineInitiator)
-            def pipelineCache = (PipelineCache) ActionDependencies.getBean(PipelineCache)
-            def pipeline = PipelineTriggerConverter.fromParameters(pipelineCache, context.parameters)
-
-            LOGGER.info("Executing PipelineTriggerAction for '${pipeline}'...")
-            pipelineInitiator.call(pipeline)
-            LOGGER.info("Successfully executed PipelineTriggerAction for '${pipeline}")
-        } catch (Exception e) {
-            LOGGER.error("Exception occurred while executing PipelineTriggerAction", e)
-            throw e
-        }
+  static final Logger LOGGER = LoggerFactory.getLogger(PipelineTriggerAction)
+
+  private final Registry registry
+
+  PipelineTriggerAction(Registry registry) {
+    this.registry = registry
+  }
+
+  @Override
+  void execute(Context context, Execution execution) throws Exception {
+    try {
+      def pipelineInitiator = (PipelineInitiator) ActionDependencies.getBean(PipelineInitiator)
+      def pipelineCache = (PipelineCache) ActionDependencies.getBean(PipelineCache)
+      def pipeline = PipelineTriggerConverter.fromParameters(pipelineCache, context.parameters)
+
+      LOGGER.info("Executing PipelineTriggerAction for '${pipeline}'...")
+      pipelineInitiator.call(pipeline)
+
+      Id id = registry.createId("pipelines.triggered")
+        .withTag("monitor", getClass().getSimpleName())
+        .withTag("application", Optional.ofNullable(pipeline.getApplication()).orElse("NULL_APPLICATION"))
```

Comment:
minor nitpick: I would go with just `"null"` instead of `"NULL_APPLICATION"`, if we want to query for it `application,null,:eq` wouldn't be too bad

Reply:
I wanted to make it clear that we were actually setting it to a value, vs null, because apparently spectator/atlas barfs on null values. Seems ok to name it something explicit (even if it's excessive caps usage) to avoid any debugging confusion of looking for a null value and finding a string "null".

Reply:
`"null"` is a pretty good string for `null` IMO, no need to get creative

Reply:
🙏